substrate/frame/election-provider-multi-block/src/signed/tests.rs

```diff
@@ -633,4 +656,648 @@ mod e2e {
+							0,
+							999,
+							ElectionScore {
+								minimal_stake: 55,
+								sum_stake: 130,
+								sum_stake_squared: 8650
+							}
+						),
+						Event::Stored(0, 999, 0),
+						Event::Stored(0, 999, 1),
+						Event::Stored(0, 999, 2),
+						Event::Slashed(0, 99, 5),
+					]
+				);
+				// Verify no additional events occurred
+				let remaining_events = verifier_events_since_last_call();
+				assert_eq!(remaining_events, vec![], "No additional verifier events should occur");
+
+				// Verifier should be STOPPED when transitioning to unsigned
+				assert_eq!(VerifierPallet::status(), crate::verifier::Status::Nothing);
+
+				// Verify no solution was queued (verification was stopped, not completed)
+				assert_eq!(VerifierPallet::queued_score(), None);
+			});
+	}
+
+	#[test]
+	fn not_all_solutions_verified_signed_verification_incomplete_to_signed() {
+		// Test that demonstrates early failure and no verifier restart due to insufficient blocks:
+		// - SignedValidation phase of 3 blocks with 3-page solutions
+		// - Bad solution with high score fails early on page 1 with invalid voter index
+		// - Good solution with lower score is submitted but cannot be verified
+		// - After bad solution fails, verifier is NOT restarted because there are only 2 blocks
+		//   remaining and we need 3 blocks for a 3-page solution
+		// - System transitions back to Signed phase with good solution still queued for next round
+		ExtBuilder::signed()
+			.pages(3)
+			.signed_validation_phase(3) // 3 blocks for validation
+			.are_we_done(crate::mock::AreWeDoneModes::BackToSigned) // Revert to signed if no solution
+			.build_and_execute(|| {
+				roll_to_signed_open();
+
+				// Submit bad solution with high score that will fail early during verification
+				{
+					let mut bad_score = mine_full_solution().unwrap().score;
+					bad_score.minimal_stake *= 2; // Make it higher score than good solution
+					assert_ok!(SignedPallet::register(RuntimeOrigin::signed(99), bad_score));
+
+					// Submit valid empty solution for page 0
+					assert_ok!(SignedPallet::submit_page(
+						RuntimeOrigin::signed(99),
+						0,
+						Some(Box::new(Default::default()))
+					));
+					// Create invalid solution with out-of-bounds voter index for page 1
+					// This will cause InvalidVoter error during feasibility check on block 2
+					let invalid_solution = crate::mock::TestNposSolution {
+						votes1: vec![(9999, 0)], // Invalid voter index 9999
+						..Default::default()
+					};
+					assert_ok!(SignedPallet::submit_page(
+						RuntimeOrigin::signed(99),
+						1,
+						Some(Box::new(invalid_solution))
+					));
+					// Submit valid empty solution for page 2
+					assert_ok!(SignedPallet::submit_page(
+						RuntimeOrigin::signed(99),
+						2,
+						Some(Box::new(Default::default()))
+					));
+				}
+
+				// Submit good solution with lower score (all pages valid)
+				{
+					let good_solution = mine_full_solution().unwrap();
```

Comment:
a good follow-up for me/you to look at is ensuring that the signed queue is FIFO. If so, if we can submit solutions of
the same score, and the first one will be verified first.